src/main/java/com/ooml_codegen/models/Enum.java

```diff
@@ -0,0 +1,117 @@
+package com.ooml_codegen.models;
+
+import com.ooml_codegen.compiler.generator.GeneratorType;
+import com.ooml_codegen.compiler.generator.enums.GenerationContext;
+import com.ooml_codegen.compiler.generator.interfaces.IGeneration;
+import com.ooml_codegen.models.enums.modifiers.access.EnumAccessModifier;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Enum implements IGeneration {
+    private final String name;
+    private final Package ePackage;
+    private final EnumAccessModifier accessModifier;
+    private final List<EnumProperty> enumerations = new ArrayList<>();
+
+    private boolean withValue = false;
+    private boolean withoutValue = false;
```

Comment:
_Si c'est trop complique pour toi pas besoin de le faire._

Possible d'utiliser seulement 1 boolean qui permet de dire si l'enum a une valeur ou non ?

ex: true = withValue // false = without

```suggestion
    private boolean isEnumHasValue = false;
``` 

Cela permet de simplifier l'utilisation des setters, des boolean methodes, ainsi que pour le template velocity.

A tester dans le template Velocity si le syntaxe `if ( !$isEnumHasValue ) { }` donne le meme resultat.